### PR TITLE
Vecdb fix

### PR DESCRIPTION
**Pull request by cwastche**

### CODE REVIEW

File: overlore/sqlite/vector_db.py [hunk 1]
```diff
@@ -98,61 +112,39 @@ def get_townhalls_from_events(self, event_ids: list[int]) -> tuple[list[str], li
...
-                event_id,
-                summary
-            FROM
-                DupDiscussions
-            WHERE
-                dup_rn = 1
+        if len(event_ids) == 0:
+            return ([], [])
+
+        event_id_placeholders = (", ".join(["(?)"] * len(event_ids))).rstrip(",")
```
Comment (edisontim) [inline overlore/sqlite/vector_db.py:119]: `rstrip` is unnecessary, `join` only adds the separator in the middle of the members it joined

Comment (edisontim) [reply overlore/sqlite/vector_db.py:119]: so no separator will be a trailing character

File: overlore/sqlite/vector_db.py [hunk 2]
```diff
@@ -98,61 +112,39 @@ def get_townhalls_from_events(self, event_ids: list[int]) -> tuple[list[str], li
...
-                SELECT
-                    json_each.value AS event_id,
-                    T.summary,
-                    T.ts,
-                    T.rowid,
-                    ROW_NUMBER() OVER (PARTITION BY json_each.value ORDER BY T.ts DESC) as rn
-                FROM
-                    townhall T, json_each(T.events_ids)
-                WHERE
-                    json_each.value IN ((?), (?), (?), (?), (?)) AND json_each.value <> 0
```
Comment (edisontim) [inline overlore/sqlite/vector_db.py:115]: `json_each.value <> 0` is this check not necessary anymore with the new method?

**Review by edisontim — CHANGES_REQUESTED** [review]
Test cases include: 
0 event_ids, 0 townhalls
1 events_ids, 1-2-3 townhalls
0 events_ids, 1-2-3 townhalls ?

File: overlore/sqlite/vector_db.py [hunk 3]
```diff
@@ -73,23 +73,37 @@ def insert_townhall_discussion(
         return rowid
 
     def query_nearest_neighbour(self, query_embedding: str, realm_id: int, limit: int = 1) -> list[StoredVector]:
+        if self.get_entries_count() == (0, 0):
+            raise RuntimeError("Empty vector db")
```
[the remark this answers is not in the document]
Comment (cwastche) [reply overlore/sqlite/vector_db.py:77]: The check seemed necessary as running it on a empty db results in a crash: "tests/test_vector_db.py .Fatal Python error: Aborted"

Returning an empty list works as well thought

File: overlore/sqlite/vector_db.py (hunk 1 above)
Comment (cwastche) [reply overlore/sqlite/vector_db.py:119]: Yep, missed that one. Removed rstrip.

File: overlore/sqlite/vector_db.py (hunk 2 above)
Comment (cwastche) [reply overlore/sqlite/vector_db.py:115]: Not necessary anymore as we do not extend the event_ids using x amounts of '0' anymore, but adapt the query to the nb of given event_ids.

**Review by edisontim — APPROVED** [review]
lgtm

**Pull request merged**